### PR TITLE
Task #11

### PR DESCRIPTION
**Pull request by fernandogutierrez**
- Modify client API methods to get, delete and post clients
- Add try/except to control exception in your  DBRedisConnector 
- Add unit tests for the client endpoints

### CODE REVIEW

File: trucks_project/api/main.py [hunk 1]
```diff
@@ -1,63 +1,48 @@
...
+    if result == 0:
+        abort(404, "Client not found!.")
+    return {"status": "Success", "message": f"Client id [ {client_id} ] deleted from DB."}
 
-@app.errorhandler(404)
-def resource_not_found(e):
-    return jsonify(err=str(e)), 400
 
+# app.add_url_rule("/clients", view_func=client_api.get_all_clients)
+# app.add_url_rule("/clients", view_func=client_api.save_client, methods=["POST"])
```
Comment (favipcj) [inline trucks_project/api/main.py]: it would be better to delete these commented lines

File: trucks_project/api/main.py [hunk 2]
```diff
@@ -1,63 +1,48 @@
...
-    for book in books:
-        if book.get("id") == int(book_id):
-            book_to_return = book
-            break
-    return jsonify(book_to_return)
-
+@app.route(f"{API_NAME}/client", methods=["POST"])
+def save_client():
+    client_params = json.dumps(request.json)
+    id = generate_random_uuid()
```
Comment (favipcj) [inline trucks_project/api/main.py:13]: if this is generated randomly how do you know which is the id of that client ? or is this just an example?

File: trucks_project/api/main.py [hunk 3]
```diff
@@ -1,63 +1,48 @@
...
+    result = ClientManager().save_document(id, client_params)
+    return {"status": "Success", "id": id} if result else {"status": "Failed"}
```
Comment (favipcj) [inline trucks_project/api/main.py]: when doing this, even when you are returning a failure inside the status you are returning a `200`  success code

File: trucks_project/api/main.py (hunk 2 above)
Comment (fernandogutierrez) [reply trucks_project/api/main.py:13]: The POST request returns which ID was generated by this method, we need to create unique objects when sending to the API.

File: trucks_project/api/main.py (hunk 1 above)
Comment (fernandogutierrez) [reply trucks_project/api/main.py]: Commented code was removed

File: trucks_project/api/main.py (hunk 3 above)
Comment (fernandogutierrez) [reply trucks_project/api/main.py]: I resolved this comments. please check.

**Pull request merged**